cmd/machine-config-daemon/start.go

```diff
@@ -56,8 +57,30 @@ func runStartCmd(cmd *cobra.Command, args []string) {
 		glog.Fatalf("error creating clients: %v", err)
 	}
 
+	// Chroot into rootfs
+	cwdString, err := os.Getwd()
+	if err != nil {
+		glog.Fatalf("error getting current workdir: %s", err)
+	}
+
+	// Ensure that the rootMount exists
+	if _, err := os.Stat(startOpts.rootMount); err != nil {
+		if os.IsNotExist(err) {
+			glog.Fatalf("rootMount %s does not exist", startOpts.rootMount)
+		}
+		glog.Fatalf("unable to verify rootMount %s exists: %s", startOpts.rootMount, err)
+	}
+
+	defer os.Chdir(cwdString)
+	// Chroot into the root file system
+	glog.Infof(`chrooting into rootMount`, startOpts.rootMount)
+	if err := syscall.Chroot(startOpts.rootMount); err != nil {
```

Comment:
You also need to `chdir` after `chroot`, otherwise your working directory could be outside of your chroot.